addons/udes_stock/models/stock_picking.py

```diff
@@ -299,6 +299,18 @@ def action_done(self):
         # Prevent recomputing the batch stat
         batches = mls.mapped("picking_id.batch_id")
         next_pickings = self.mapped("u_next_picking_ids")
+
+        # If operation type does not allow multiple lots in a single transfer then check for number 
+        # of different lots in move lines and raise an error if more than one
+        restrict_multi_lot_pickings = self.filtered(
+            lambda p: p.picking_type_id.u_restrict_multi_lot_pickings
+        )
+
+        for pick in restrict_multi_lot_pickings:
+            lots = pick.move_line_ids.mapped("lot_id")
```

Comment:
One thing to bear in mind is that if we are tracking by serial number rather than lots then this will pull out all the serial numbers as well. So we either need to decide if we want to filter out products tracked by serial or edit the messages to make it clear that u_restrict_multi_lot_pickings applies to serial numbers as well

Reply:
Hmm good point, one for me to check with product most likely. :)

Reply:
Probably getting it to apply to both would be easier and give more functionality, would just require changing messages and possible the config name to say 'tracking' or 'lot/serial' instead of 'lot'. May be best to check with product though

Reply:
One other thing with this is if we have one move line without tracking and one with tracking then the length would be 1 and this would pass. Is this what we want?

Reply:
True, `mapped` won't return any False values. I guess there's not a way to force it to do this so might need a different approach here.

Reply:
This could be worth adding as another test actually. Picking with lot and non-lot tracked products.

Reply:
I will go check that other point with product also. I would assume we would want to count no lot as 1.

Reply:
After speaking with product:

1. This should only apply to products tracked by lot, as otherwise for serial numbers we would end up with cases where products need to be moved one by one.

2. No lot should count as 1 for the lot cout, so this will need to be adjusted.

Reply:
Have now pushed a change so that a lot restricted picking cannot contain a move line with a lot and then another with no lot set (basically for a non-lot tracked product).